iceoryx_utils/source/posix_wrapper/shared_memory_object/memory_map.cpp

```diff
@@ -67,44 +49,92 @@ MemoryMap::MemoryMap(const void* f_baseAddressHint,
-#if defined(QNX) || defined(QNX__) || defined(__QNX__)
-    int l_lockflags = MCL_CURRENT;
-
-    auto mlockCall = cxx::makeSmartC(mlockall, cxx::ReturnMode::PRE_DEFINED_ERROR_CODE, {-1}, {}, l_lockflags);
-
-    if (mlockCall.hasErrors())
-    {
-        std::cerr << "Failed to perform memory mapping : " << mmapCall.getErrorString() << std::endl;
-        m_isLocked = false;
-    }
-    else
+MemoryMapError MemoryMap::errnoToEnum(const int32_t errnum) const noexcept
+{
+    switch (errnum)
     {
-        m_isLocked = true;
-    }
-#endif
-} // namespace posix
-
+    case EACCES:
+        std::cerr
+            << "One or more of the following failures happened:\n"
+            << "  1. The file descriptor belongs to a non-regular file.\n"
+            << "  2. The file descriptor is not opened for reading.\n"
+            << "  3. MAP_SHARED is requested and PROT_WRITE is set but the file descriptor is not opened for writing.\n"
+            << "  4. PROT_WRITE is set but the file descriptor is set to append-only." << std::endl;
+        return MemoryMapError::ACCESS_FAILED;
+    case EAGAIN:
+        std::cerr << "Either to much memory has been locked or the file is already locked." << std::endl;
+        return MemoryMapError::UNABLE_TO_LOCK;
+    case EBADF:
+        std::cerr << "Invalid file descriptor provided." << std::endl;
+        return MemoryMapError::INVALID_FILE_DESCRIPTOR;
+    case EEXIST:
+        std::cerr << "The mapped range that is requested is overlapping with an already mapped memory range."
+                  << std::endl;
+        return MemoryMapError::MAP_OVERLAP;
+    case EINVAL:
+        std::cerr << "One or more of the following failures happened:\n"
+                  << "  1. The address, length or the offset is not aligned on a page boundary.\n"
+                  << "  2. The provided length is 0.\n"
+                  << "  3. One of the flags of MAP_PRIVATE, MAP_SHARED or MAP_SHARED_VALIDATE is missing." << std::endl;
+        return MemoryMapError::INVALID_PARAMETERS;
+    case ENFILE:
+        std::cerr << "System limit of maximum open files reached" << std::endl;
+        return MemoryMapError::OPEN_FILES_SYSTEM_LIMIT_EXCEEDED;
+    case ENODEV:
+        std::cerr << "Memory mappings are not supported by the underlying filesystem." << std::endl;
+        return MemoryMapError::FILESYSTEM_DOES_NOT_SUPPORT_MEMORY_MAPPING;
+    case ENOMEM:
+        std::cerr << "One or more of the following failures happened:\n"
+                  << "  1. Not enough memory available.\n"
+                  << "  2. The maximum supported number of mappings is exceeded.\n"
+                  << "  3. Partial unmapping of an already mapped memory region dividing it into two parts.\n"
+                  << "  4. The processes maximum size of data segments is exceeded.\n"
+                  << "  5. The sum of the number of pages and offset are overflowing. (only 32-bit architecture)"
```

Comment:
I do not understand what 5 is supposed to mean. Is it because the total memory requested does not fit into 32 bit or something like this? Maybe clarify the error output.

I cannot really believe it is the number of pages itself when we can count 2³² of those on 32 bit systems, using all of them would mean more memory than is really available on real systems given a page size of 4096 bytes. Of course virtual memory changes that... 

Regardless, I am not sure what is meant to be overflowing here.

Reply:
It is something from the man pages which I added here for completeness. I rewrote it a bit so that it is more detailed.